djangosaml2/conf.py

```diff
@@ -51,13 +53,79 @@ def get_config_loader(path, request=None):
     return config_loader
 
 
+def config_map(data):
+    return {
+        # full path to the xmlsec1 binary programm
+        'xmlsec_binary': '/usr/bin/xmlsec1',
+
+        # your entity id, usually your subdomain plus the url to the metadata view
+        'entityid': data.idp_entity,
+
+        # directory with attribute mapping
+        "attribute_map_dir": data.attributes_dir.name,
+
+        # this block states what services we provide
+        'service': {
+            # we are just a lonely SP
+            'sp': {
+                # 'name': 'Federated Django sample SP',
+                'name_id_format': getattr(saml2.saml, data.name_id_format,
+                                          saml2.saml.NAMEID_FORMAT_UNSPECIFIED),
+
+                # For Okta add signed logout requets. Enable this:
+                # "logout_requests_signed": True,
+
+                'endpoints': {
+                    # url and binding to the assertion consumer service view
+                    # do not change the binding or service name
+                    'assertion_consumer_service': [
+                        (data.acs_uri,
+                        saml2.BINDING_HTTP_POST),
+                    ],
+                    # url and binding to the single logout service view
+                    # do not change the binding or service name
+                    'single_logout_service': [
+                        (data.single_logout_service_uri_redirect,
+                        saml2.BINDING_HTTP_REDIRECT),
+                        (data.single_logout_service_uri_post,
+                        saml2.BINDING_HTTP_POST),
+                    ],
+                },
+            },
+        },
+
+        # where the remote metadata is stored, local, remote or mdq server.
+        # One metadatastore or many ...
+        'metadata': {
+            'local': [data.metadata_file.name],
+        },
+
+        'debug': 1,
+
+        # Signing
+        'key_file': data.sp_key_file.name,  # private part
+        'cert_file': data.sp_certificate_file.name,  # public part
+
+        # Encryption
+        'encryption_keypairs': [{
+            'key_file': data.sp_key_file.name,  # private part
+            'cert_file': data.sp_certificate_file.name,  # public part
+        }]
+    }
+
+
 def config_settings_loader(request=None):
     """Utility function to load the pysaml2 configuration.
 
     This is also the default config loader.
     """
-    conf = SPConfig()
-    conf.load(copy.deepcopy(settings.SAML_CONFIG))
+    conf = saml2.config.SPConfig()
+    saml_config_list = list(SamlConfig.objects.all())
```

Comment:
the list is not required here I guess.

Reply:
here we can simply do `saml_config_list = SamlConfig.objects.first()`

Reply:
removed list and used Shivin's approach.